### **📌 Objective – Level 3, Task 2: Customer Preference Analysis**  

In this task, we aim to **understand customer preferences by analyzing cuisine types and their impact on restaurant ratings and popularity**.  

### **🎯 Key Goals:**  
1️⃣ **Analyze the relationship** between **cuisine type and restaurant ratings**.  
2️⃣ **Identify the most popular cuisines** based on the **number of votes** received.  
3️⃣ **Determine if certain cuisines consistently receive higher ratings** than others.  

This analysis will help us:  
✔ Discover which cuisines customers **prefer the most**.  
✔ Understand how **cuisine type affects restaurant success**.  
✔ Identify potential **trends in customer dining n you're ready! 😊

In [9]:
import pandas as pd 
df = pd.read_csv("leve2_dataset.csv")

### Step 1: Analyze the Relationship Between Cuisine Type and Ratings
In this step, we will examine whether certain cuisines receive higher ratings than others.



In [11]:
# Calculate the average rating for each cuisine type
avg_rating_by_cuisine = df.groupby('Cuisines')['Aggregate rating'].mean().sort_values(ascending=False)

# Print results
print(avg_rating_by_cuisine.head(10))  # Display top 10 highest-rated cuisines


Cuisines
Italian, Deli               4.9
Hawaiian, Seafood           4.9
American, Sandwich, Tea     4.9
Continental, Indian         4.9
European, Asian, Indian     4.9
European, Contemporary      4.9
European, German            4.9
BBQ, Breakfast, Southern    4.9
American, Coffee and Tea    4.9
Sunda, Indonesian           4.9
Name: Aggregate rating, dtype: float64


### Step 2: Identify the Most Popular Cuisines Based on Votes
Now, let’s check which cuisines are the most popular among customers by analyzing the number of votes received.

In [14]:
# Find the most popular cuisines based on the total number of votes
popular_cuisines = df.groupby('Cuisines')['Votes'].sum().sort_values(ascending=False)

# Print results
print(popular_cuisines.head(10))  # Display top 10 most voted cuisines


Cuisines
North Indian, Mughlai             53747
North Indian                      46241
North Indian, Chinese             42012
Cafe                              30657
Chinese                           21925
North Indian, Mughlai, Chinese    20115
Fast Food                         17852
South Indian                      16433
Mughlai, North Indian             15275
Italian                           14799
Name: Votes, dtype: int64


### Step 3: Determine If Certain Cuisines Receive Higher Ratings
Now, let’s check whether the most popular cuisines also tend to receive higher ratings.

In [17]:
# Get the average rating for the top-voted cuisines
popular_cuisine_ratings = df.groupby('Cuisines')[['Votes', 'Aggregate rating']].mean().sort_values(by='Votes', ascending=False)

# Print results
print(popular_cuisine_ratings.head(10))  # Display top 10 cuisines by votes with their average rating


                                                           Votes  \
Cuisines                                                           
Italian, American, Pizza                            10934.000000   
American, Burger, Cafe                               9667.000000   
Finger Food, North Indian, Italian, Continental...   5705.000000   
American, Continental, Italian, North Indian, A...   4914.000000   
European, Mediterranean, North Indian                4180.333333   
Cafe, Mexican, American, Italian, Lebanese, Con...   4085.000000   
American, North Indian, European, Asian              3697.000000   
North Indian, Mediterranean, Asian, Continental      3413.000000   
American, Desserts, Steak                            3157.000000   
Desserts, Bar Food                                   3074.000000   

                                                    Aggregate rating  
Cuisines                                                              
Italian, American, Pizza                 

### **📊 Level 3 – Task 2: Customer Preference Analysis**  

---

### **📌 Objective**  
The goal of this task is to analyze **customer preferences** by studying the relationship between **cuisine types, restaurant ratings, and customer engagement** (measured by votes).  

### **🎯 Key Goals:**  
1️⃣ **Analyze the relationship** between **cuisine type and restaurant ratings**.  
2️⃣ **Identify the most popular cuisines** based on the **number of votes received**.  
3️⃣ **Determine if certain cuisines consistently receive higher ratings** than others.  

This analysis helps in understanding:  
✔ **Which cuisines are most loved by customers**.  
✔ **How cuisine type affects restaurant success and customer satisfaction**.  
✔ **Whether popularity (votes) translates to better ratings**.  

---

## **1️⃣ Step 1: Relationship Between Cuisine Type and Ratings**  
### **🔹 Process:**  
To identify which cuisines receive higher ratings, we calculated the **average restaurant rating** for each cuisine.  

### **📊 Highest-rated cuisines (average rating = 4.9):**  
✔ **(Your highest-rated cuisines, e.g., Italian, Deli, Hawaiian, Seafood, etc.)**  

### **🔹 Key Insights:**  
✔ **Fusion cuisines (e.g., European-Asian, Continental-Indian) tend to have high ratings.**  
✔ **Beverage-focused cuisines** (e.g., Coffee and Tea) also receive **high ratings**, indicating strong customer preference.  
✔ **International cuisines** like Italian and European appear frequently in **top-rated categories**.  

---

## **2️⃣ Step 2: Most Popular Cuisines Based on Votes**  
### **🔹 Process:**  
We analyzed the total **number of votes** each cuisine received to determine which cuisines are most popular among customers.  

### **📊 Most popular cuisines (highest votes received):**  
✔ **(Your most voted cuisines, e.g., North Indian, Chinese, Fast Food, etc.)**  

### **🔹 Key Insights:**  
✔ **Cuisines with the highest votes indicate strong customer engagement and preference.**  
✔ **Popularity doesn’t always mean higher ratings** – some widely available cuisines may have lower ratings.  
✔ **High-voted cuisines are likely influenced by availability, affordability, or cultural trends.**  

---

## **3️⃣ Step 3: Comparing Ratings of Popular Cuisines**  
### **🔹 Process:**  
We compared the **average rating of the most voted cuisines** to check whether the most popular cuisines also tend to receive **higher ratings**.  

### **📊 Findings:**  
✔ **(Your result – Do the most popular cuisines also have high ratings or not?)**  

### **🔹 Key Insights:**  
✔ **Some cuisines are highly rated but not widely popular**, meaning they cater to a **niche audience**.  
✔ **High-voted cuisines don’t always have the best ratings**, suggesting **high demand but mixed customer satisfaction**.  
✔ **Cuisines that have both high votes and high ratings** can be considered **the most successful in terms of quality and popularity**.  

---

### **🎯2Final Summary**  
✔ **Successfully completed Level 3 – Task 1: Customer Preference Analysis.**  
✔ **Extracted key insights on cuisine ratings, popularity, and customer behavior.**  
✔ **Learned that high votesmentation.**  
🚀 **Let me know if you’re ready to start Task 2!** 😊mentation.**  
🚀 **Let me know if you’re ready to start Task 2!** 😊